services/core-web/src/components/dashboard/noticeOfWorkHomePage/NoticeOfWorkHomePage.js

```diff
@@ -35,58 +33,62 @@ const propTypes = {
   applicationTypeOptions: CustomPropTypes.options.isRequired,
 };
 
-export class NoticeOfWorkHomePage extends Component {
-  params = queryString.parse(this.props.location.search);
-
-  listQueryParams = [];
-
-  splitListParams = formatQueryListParams("split", this.listQueryParams);
-
-  joinListParams = formatQueryListParams("join", this.listQueryParams);
+const defaultParams = {
+  page: Strings.DEFAULT_PAGE,
+  per_page: Strings.DEFAULT_PER_PAGE,
+  sort_field: "received_date",
+  sort_dir: "desc",
+  mine_search: undefined,
+  now_number: undefined,
+  mine_name: undefined,
+  originating_system: [],
+  lead_inspector_name: undefined,
+  now_application_status_description: [],
+  notice_of_work_type_description: [],
+  mine_region: [],
+};
 
-  // Holds list params as array
+export class NoticeOfWorkHomePage extends Component {
   state = {
     isLoaded: false,
-    params: {
-      page: Strings.DEFAULT_PAGE,
-      per_page: Strings.DEFAULT_PER_PAGE,
-      ...this.params,
-    },
+    params: defaultParams,
   };
 
   componentDidMount() {
-    const params = this.props.location.search;
-    const parsedParams = queryString.parse(params);
-    const { page = this.state.params.page, per_page = this.state.params.per_page } = parsedParams;
-    if (params) {
-      this.renderDataFromURL();
-    } else {
-      this.props.history.replace(
-        router.NOTICE_OF_WORK_APPLICATIONS.dynamicRoute({
-          page,
-          per_page,
-        })
-      );
-    }
+    const params = queryString.parse(this.props.location.search);
+    this.setState(
+      (prevState) => ({
+        params: {
+          ...prevState.params,
+          ...params,
+        },
+      }),
+      () =>
+        this.props.history.replace(
+          routes.NOTICE_OF_WORK_APPLICATIONS.dynamicRoute(this.state.params)
+        )
+    );
   }
 
   componentWillReceiveProps(nextProps) {
-    const locationChanged = nextProps.location !== this.props.location;
-    if (locationChanged) {
-      this.renderDataFromURL(nextProps.location.search);
+    if (nextProps.location !== this.props.location) {
+      this.setState({ isLoaded: false }, () => this.renderDataFromURL(nextProps.location.search));
     }
   }
 
-  componentWillUnmount() {
-    this.setState({ params: {} });
-  }
+  renderDataFromURL = (params) => {
+    const parsedParams = queryString.parse(params);
+    this.props.fetchNoticeOfWorkApplications(parsedParams).then(() => {
+      this.setState({ isLoaded: true });
+    });
+  };
 
   renderDataFromURL = (queryParams) => {
     const params = queryParams || this.props.location.search;
```

Comment:
was this removed? Seems the same function was added above

Reply:
This was the one push I meant to make before you reviewed, I forgot to delete the old version of the function (but everything still worked as intended since the new one was defined above the old one).